### PR TITLE
[ls-qpack] Add cmake export

### PR DESCRIPTION
**Pull request by talregev**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by talregev** [comment]
I try to add export cmake and it create an error that I don't know how to solve. Help will be appreciate. 

```
CMake Error in CMakeLists.txt:
  Target "ls-qpack" INTERFACE_INCLUDE_DIRECTORIES property contains path:

    "C:/src/vcpkg/buildtrees/ls-qpack/src/v2.5.4-e6ccc134ae.clean/."

  which is prefixed in the source directory.


CMake Error in CMakeLists.txt:
  Target "ls-qpack" INTERFACE_INCLUDE_DIRECTORIES property contains path:

    "C:/src/vcpkg/buildtrees/ls-qpack/src/v2.5.4-e6ccc134ae.clean/wincompat"

  which is prefixed in the source directory.
```

**Comment by talregev** [comment]
I solve the error

**Review by dg0yt — COMMENTED** [review]
I can't check the details now, but I would expect that the exported config needs a `find_dependency` for xxhash at least for static linkage - that's also what we learned from the msh PR.

**Comment by MonicaLiu0311** [comment]
usage:
```
ls-qpack provides CMake targets:

  # this is heuristically generated, and may not be correct
  find_package(unofficial-ls-qpack CONFIG REQUIRED)
  target_link_libraries(main PRIVATE unofficial::ls-qpack::ls-qpack)
```

When testing usage, the following error occurs:
```
1> [CMake] -- Configuring done (3.6s)
1> [CMake] CMake Error at E:/ls-qpack/installed/x64-windows/share/unofficial-ls-qpack/unofficial-ls-qpack-config.cmake:60 (set_target_properties):
1> [CMake]   The link interface of target "unofficial::ls-qpack::ls-qpack" contains:
1> [CMake] 
1> [CMake]     PkgConfig::XXH
1> [CMake] 
1> [CMake]   but the target was not found.  Possible reasons include:
1> [CMake] 
1> [CMake]     * There is a typo in the target name.
1> [CMake]     * A find_package call is missing for an IMPORTED target.
1> [CMake]     * An ALIAS target is missing.
1> [CMake] 
1> [CMake] Call Stack (most recent call first):
1> [CMake]   E:/ls-qpack/scripts/buildsystems/vcpkg.cmake:859 (_find_package)
1> [CMake]   CMakeProject/CMakeLists.txt:14 (find_package)
```

<details>
<summary><b>test.cpp</b></summary>
<pre>
#include &#60;iostream&#62;
#include "lsqpack.h"

using namespace std;

int main()
{
        cout << "Hello CMake." << endl;
        return 0;
}
</pre></details>


<details>
<summary><b>CMakeLists.txt</b></summary>
<pre>
cmake_minimum_required (VERSION 3.8)

set(CMAKE_TOOLCHAIN_FILE "E:/ls-qpack/vcpkg/scripts/buildsystems/vcpkg.cmake")

project ("test")

add_library (main "test.cpp")

find_package(unofficial-ls-qpack CONFIG REQUIRED)
target_link_libraries(main PRIVATE unofficial::ls-qpack::ls-qpack)
</pre></details>

**Comment by talregev** [comment]
@dg0yt Do you have suggestion here?

**Comment by talregev** [comment]
@MonicaLiu0311 I change the way we discover xxhash. 
Can you test it again?

**Comment by MonicaLiu0311** [comment]
It's also missing this:
image

**Comment by talregev** [comment]
@MonicaLiu0311 
Should I create a usage file?

```
ls-qpack provides CMake targets:

  include(CMakeFindDependencyMacro)
  find_dependency(xxhash CONFIG)
  find_package(unofficial-ls-qpack CONFIG REQUIRED)
  target_link_libraries(main PRIVATE unofficial::ls-qpack::ls-qpack)
```

**Comment by talregev** [comment]
@MonicaLiu0311 I added usage file. Can you verify?

File: ports/ls-qpack/usage [hunk 1]
```diff
@@ -1,7 +1,6 @@
...
-
-    find_path(LSQPACK_INCLUDE_DIRS "lsqpack.h")
-    find_library(LSQPACK_LIBRARY ls-qpack REQUIRED)
-
-    target_include_directories(main PRIVATE ${LSQPACK_INCLUDE_DIRS})
-    target_link_libraries(main PRIVATE ${LSQPACK_LIBRARY})
+ls-qpack provides CMake targets:
+
+  include(CMakeFindDependencyMacro)
+  find_dependency(xxhash CONFIG)
```
Comment (MonicaLiu0311) [inline ports/ls-qpack/usage:4]:
```suggestion

```

File: ports/ls-qpack/portfile.cmake [hunk 2]
```diff
@@ -28,6 +29,7 @@ vcpkg_copy_pdbs()
 vcpkg_fixup_pkgconfig()
 
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
-vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
```
Comment (MonicaLiu0311) [inline ports/ls-qpack/portfile.cmake:32]:
```
file(READ "${CURRENT_PACKAGES_DIR}/share/unofficial-ls-qpack/unofficial-ls-qpack-config.cmake" cmake_config)
file(WRITE "${CURRENT_PACKAGES_DIR}/share/unofficial-ls-qpack/unofficial-ls-qpack-config.cmake"
"include(CMakeFindDependencyMacro)
find_dependency(xxhash CONFIG)
${cmake_config}
")
```

Comment (MonicaLiu0311) [reply ports/ls-qpack/portfile.cmake:32]: Add it.

Comment (talregev) [reply ports/ls-qpack/portfile.cmake:32]: Done!
Thank you for your help! :)

File: ports/ls-qpack/usage (hunk 1 above)
Comment (talregev) [reply ports/ls-qpack/usage]: I remove all the usage file.

Comment (talregev) [reply ports/ls-qpack/usage]:
```
ls-qpack provides CMake targets:

  # this is heuristically generated, and may not be correct
  find_package(unofficial-ls-qpack CONFIG REQUIRED)
  target_link_libraries(main PRIVATE unofficial::ls-qpack::ls-qpack)
```

**Review by MonicaLiu0311 — DISMISSED** [review]

**Comment by MonicaLiu0311** [comment]
The usage test passed on `x64-windows` (header files found):
```
ls-qpack provides CMake targets:

  find_package(unofficial-ls-qpack CONFIG REQUIRED)
  target_link_libraries(main PRIVATE unofficial::ls-qpack::ls-qpack)
```

**Review by dg0yt — CHANGES_REQUESTED** [review]
This port lacks `vcpkg_cmake_config_fixup`. It is mandatory for multi-config installation.

**Comment by talregev** [comment]
> This port lacks `vcpkg_cmake_config_fixup`. It is mandatory for multi-config installation.

Done.

File: ports/ls-qpack/portfile.cmake [hunk 3]
```diff
@@ -26,8 +27,18 @@ vcpkg_cmake_configure(
 vcpkg_cmake_install()
 vcpkg_copy_pdbs()
 vcpkg_fixup_pkgconfig()
+vcpkg_cmake_config_fixup()
```
Comment (dg0yt) [inline ports/ls-qpack/portfile.cmake:30]:
```suggestion
vcpkg_cmake_config_fixup(PACKAGE_NAME unofficial-ls-qpack)
```

**Review by dg0yt — CHANGES_REQUESTED** [review]
Otherwise LGTM. :+1:

File: ports/ls-qpack/portfile.cmake (hunk 3 above)
Comment (talregev) [reply ports/ls-qpack/portfile.cmake]: Done.

**Review by MonicaLiu0311 — APPROVED** [review]

**Pull request merged**